s1/Program copy.cs

```diff
@@ -0,0 +1,117 @@
+﻿int n_rows = 10;
+int n_cols = 30;
+int score = 0;
+string statusMessage = "";
+char[,] map = new char[n_rows, n_cols];
+var random = new Random();
+
+int n_treasures = (int)(n_rows * n_cols * 0.01);
+char treasure = '$';
+
+int n_obstacles = (int)(n_rows * n_cols * 0.05);
+char obstacle = 'o';
+
+
+for (var y = 0; y < n_rows; y++)
+{
+  for (var x = 0; x < n_cols; x++)
+  {
+    map[y, x] = ' ';
+  }
+}
+
+for (var i = 0; i < n_obstacles; i++)
+{
+  map[random.Next(n_rows), random.Next(n_cols)] = obstacle;
+}
+
+for (var i = 0; i < n_treasures; i++)
+{
+  map[random.Next(n_rows), random.Next(n_cols)] = treasure;
```

Comment:
no guarantee of spawning on different coordinates - you'll run into a bug when multiple treasures are spawned on the same spot.